### ISSUE TITLE
 gtk module compile error on Centos 6

### ISSUE THREAD

**Issue by sebd77** [alfredh/baresip#122]
Hi! Regarding the issue "gtk module compile error on OpenBSD 5.7 #53 ", I do have a very similar issue over Centos6. I could install properly re-0.4.15 and rem-0.4.7, however when I try to install baresip-0.4.18, the make returns the following error:
[root@t420sdlinux baresip-0.4.18]# make
CC [M] build-i686/modules/gtk/gtk_mod.o
modules/gtk/gtk_mod.c:71: error: expected â=â, â,â, â;â, âasmâ or âattributeâ before âapp_entriesâ
modules/gtk/gtk_mod.c:74: warning: ISO C does not allow extra â;â outside of a function
modules/gtk/gtk_mod.c: In function âget_call_from_gvariantâ:
modules/gtk/gtk_mod.c:84: warning: cast to pointer from integer of different size
modules/gtk/gtk_mod.c: In function ânotify_incoming_callâ:
modules/gtk/gtk_mod.c:314: error: too few arguments to function ânotify_notification_newâ
modules/gtk/gtk_mod.c: In function âmessage_handlerâ:
modules/gtk/gtk_mod.c:575: error: too few arguments to function ânotify_notification_newâ
modules/gtk/gtk_mod.c: In function âgtk_threadâ:
modules/gtk/gtk_mod.c:830: warning: implicit declaration of function âg_action_map_add_action_entriesâ
modules/gtk/gtk_mod.c:830: warning: nested extern declaration of âg_action_map_add_action_entriesâ
modules/gtk/gtk_mod.c:830: warning: implicit declaration of function âG_ACTION_MAPâ
modules/gtk/gtk_mod.c:830: warning: nested extern declaration of âG_ACTION_MAPâ
modules/gtk/gtk_mod.c:831: error: âapp_entriesâ undeclared (first use in this function)
modules/gtk/gtk_mod.c:831: error: (Each undeclared identifier is reported only once
modules/gtk/gtk_mod.c:831: error: for each function it appears in.)
make: *** [build-i686/modules/gtk/gtk_mod.o] Error 1
[root@t420sdlinux baresip-0.4.18]#
would you have idea? Not sure which package is missing? Thanks!
@sebd77

Attach files by dragging & dropping or selecting them.
Styling with Markdown is supported

**Comment by clehner** [comment]
@sebd77  what versions of glib, gtk, and libnotify do you have? Do you actually want to use the gtk module or is it just getting in the way? If you don't need it you can try compiling with `make USE_GTK=`

**Comment by sebd77** [comment]
Hi, please find below the installed versions. I don't really need the gtk module, so I just tried the commands "make USE_GTK=" and "sudo make install USE_GTK=" and both of them finished properly. It seems we are OK now. Thank you so much!
[root@t420sdlinux ~]# rpm -qa|grep "^glib"
glibc-common-2.12-1.166.el6_7.7.i686
glibc-devel-2.12-1.166.el6_7.7.i686
glibc-debuginfo-common-2.12-1.166.el6_7.7.i686
glib2-doc-2.28.8-4.el6.noarch
glib-networking-debuginfo-2.28.6.1-2.2.el6.i686
glibc-static-2.12-1.166.el6_7.7.i686
glib2-2.28.8-4.el6.i686
glib2-debuginfo-2.28.8-4.el6.i686
glibmm24-2.22.1-1.el6.i686
glib-networking-2.28.6.1-2.2.el6.i686
glibc-2.12-1.166.el6_7.7.i686
glib2-static-2.28.8-4.el6.i686
glib2-devel-2.28.8-4.el6.i686
glibc-debuginfo-2.12-1.166.el6_7.7.i686
glibc-headers-2.12-1.166.el6_7.7.i686
glibc-utils-2.12-1.166.el6_7.7.i686
[root@t420sdlinux ~]# rpm -qa|grep "^gtk"
gtkhtml2-2.11.1-7.el6.i686
gtk-vnc-python-0.3.10-3.el6.i686
gtk2-immodules-2.24.23-6.el6.i686
gtk2-debuginfo-2.24.23-6.el6.i686
gtk-doc-1.11-5.1.el6.noarch
gtkspell-2.0.16-1.el6.i686
gtk2-engines-debuginfo-2.18.4-5.el6.centos.i686
gtkmm24-2.18.2-1.el6.i686
gtksourceview2-2.8.2-4.el6.i686
gtk-vnc-0.3.10-3.el6.i686
gtk2-immodule-xim-2.24.23-6.el6.i686
gtk2-engines-2.18.4-5.el6.centos.i686
gtk2-engines-devel-2.18.4-5.el6.centos.i686
gtk2-devel-docs-2.24.23-6.el6.i686
gtk2-2.24.23-6.el6.i686
gtk2-devel-2.24.23-6.el6.i686
[root@t420sdlinux ~]# rpm -qa|grep "libnotify"
libnotify-devel-0.5.0-1.el6.i686
libnotify-0.5.0-1.el6.i686
[root@t420sdlinux ~]#

**Comment by alfredh** [comment]
@clehner some of the APIs used are only available in recent versions, e.g.:

```
static GActionEntry app_entries[] = {
	{"answer", answer_activated, "x", NULL, NULL, {0} },
	{"reject", reject_activated, "x", NULL, NULL, {0} },
};

```

https://developer.gnome.org/gio/stable/GActionMap.html

Since: 2.32


perhaps we should add some checks for this?

for example this check is very nice:

```
ifneq ($(USE_GTK),)
USE_LIBNOTIFY := $(shell pkg-config 'libnotify glib-2.0 < 2.40' && echo "yes")
endif
```

**Comment by clehner** [comment]
@alfredh yes, checking with pkg-config would fix this

**Comment by alfredh** [comment]
I'm afraid I had to revert  the last commit.

building fails on Debian 7.x:

modules/gtk/gtk_mod.c:11:21: fatal error: gtk/gtk.h: No such file or directory

**Comment by alfredh** [comment]
I guess we also need to check for gtk+ ?

is it possible to do something like this (instead, or also:)

```
pkg-config 'gtk+-2.0 >= 2.22' 
```

**Comment by clehner** [comment]
> `pkg-config 'gtk+-2.0 >= 2.22'`

sounds good. I think that will cover the needed glib requirement.

**Comment by alfredh** [comment]
should be fixed in cf502319c57e67d5321882dcf5a1f15c7207e041
